### PR TITLE
Remove hardcoded values in estimate_fee test

### LINKED ISSUE #1368: dev: Better estimate fee test
Instead of comparing hardcoded values, execute the transaction and compare the actual fees with the estimated fees

### PR DESCRIPTION
**Pull request by fishseabowl**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Testing

## What is the current behavior?

Resolves: #1368 

## What is the new behavior?

Replace hardcoded value in estimate_fee test with actual fee

### CODE REVIEW

File: starknet-rpc-test/estimate_fee.rs [hunk 1]
```diff
@@ -103,13 +105,30 @@ async fn works_ok(madara: &ThreadSafeMadaraClient) -> Result<(), anyhow::Error>
...
+        calldata,
+        is_query: false,
+    };
+
+    let invoke_transaction = BroadcastedTransaction::Invoke(tx.clone());
+
+    let invoke_transaction_2 =
+        BroadcastedTransaction::Invoke(BroadcastedInvokeTransaction { nonce: FieldElement::ONE, ..tx });
+
+    let simulations = rpc
```
Comment (apoorvsadana) [inline starknet-rpc-test/estimate_fee.rs:122]: can we execute it instead of doing the simulation? i think that would be a better test, wdyt?

Comment (fishseabowl) [reply starknet-rpc-test/estimate_fee.rs:122]: @apoorvsadana Thank you for your comment. Could you please clarify how to check the actual fee after execution? Additionally, does "executing it" refer to calling JsonRpcClient.add_invoke_transaction? Thanks

Comment (apoorvsadana) [reply starknet-rpc-test/estimate_fee.rs:122]: You can fetch the receipt to get the actual_fee. Yes, you need to call add_invoke_transaction

Comment (fishseabowl) [reply starknet-rpc-test/estimate_fee.rs]: @apoorvsadana, Thank you for your help. do you know how long we should wait until `MaybePendingTransactionReceipt` gets the `Receipt` value? In my local test case, waiting for 20 seconds works fine. However, in the [rpc-tests](https://github.com/keep-starknet-strange/madara/actions/runs/8590075466/job/23537285842?pr=1526), it fails.

**Comment by tdelabro** [comment]
blocked by #1573

We are going to wait a bit before rebasing and merging this one, okay?

**Comment by fishseabowl** [comment]
@apoorvsadana Could you help to review this PR? Thanks

**Comment by tdelabro** [comment]
repository archived in favor of https://github.com/madara-alliance/madara

**Pull request closed without merging**